examples/MNIST_classification.ipynb

```diff
@@ -0,0 +1,598 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Case study: MNIST hand-written digits dataset\n",
+    "\n",
+    "##### License: Apache 2.0\n",
+    "\n",
+    "\n",
+    "This notebook shows how to use *giotto-tda* to generate features for classifying digits. We first show how to build a few topological features and present a pipeline extracting a very large amount of features for classification.\n",
+    "\n",
+    "The MNIST database of handwritten digits with 784 features, raw data available at: http://yann.lecun.com/exdb/mnist/. It can be split in a training set of the first 60,000 examples, and a test set of 10,000 examples."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import libraries\n",
+    "The first step consists in importing relevant *gtda* components and other useful libraries or modules."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "_cell_guid": "b1076dfc-b9ad-4769-8c92-a6c4dae69d19",
+    "_uuid": "8f2839f25d086af736a60e9eeb907d3b93b6e0e5"
+   },
+   "outputs": [],
+   "source": [
+    "from gtda.images import Binarizer, Inverter, ImageToPointCloud, HeightFiltration, DilationFiltration, RadialFiltration, ErosionFiltration, SignedDistanceFiltration\n",
+    "from gtda.homology import CubicalPersistence\n",
+    "from gtda.diagrams import ForgetDimension, Amplitude, Scaler, PersistenceEntropy, BettiCurve, PersistenceLandscape, HeatKernel, Silhouette\n",
+    "from gtda.plotting import plot_heatmap, plot_betti_curves, plot_diagram\n",
+    "from sklearn.pipeline import Pipeline, make_pipeline, FeatureUnion, make_union\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.model_selection import GridSearchCV\n",
+    "from sklearn.metrics import classification_report\n",
+    "import numpy as np\n",
+    "\n",
+    "# This would be quite nice but is available with sklearn >= 0.23\n",
```

Comment:
I like the idea of putting this. We could still put an if statement detecting the sklearn version, so it works for everybody.

Reply:
I think giotto-tda should update the required version of sklearn at each release. The tight integration to sklearn demands that we are always aligned with its latest version.

Reply:
An interesting suggestion. What are the downsides of imposing a specific version instead of a minimum version number?

Reply:
Backward compatibility that requires us to maintain our code should we use a sklearn feature that has become deprecated. I don't see any other.

I think my previous comment applies to all of our dependencies. In general, new releases improve performances and fix bugs, so it is useful to make use of that. It may require some work though.